QModManager/API/SMLHelper/Utility/ItemStorageHelper.cs

```diff
@@ -0,0 +1,332 @@
+﻿namespace QModManager.API.SMLHelper.Utility
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// A utility class that offers additional info about <see cref="ItemsContainer"/> instances.
+    /// </summary>
+    /// <seealso cref="IStorageHelper" />
+    public class ItemStorageHelper : IStorageHelper
+    {
+        private readonly Dictionary<ItemsContainer, Dictionary<Vector2int, bool>> HasRoomCacheCollection = new Dictionary<ItemsContainer, Dictionary<Vector2int, bool>>();
+
+        internal static readonly ItemStorageHelper Singleton = new ItemStorageHelper();
+
+        public static IStorageHelper Main => Singleton;
+
+        private ItemStorageHelper()
+        {
+            // Hides contructor
+        }
+
+        #region Common Item Sizes
+
+        private static readonly Vector2int Size1x1 = new Vector2int(1, 1);
+        private static readonly Vector2int Size1x2 = new Vector2int(1, 2);
+        private static readonly Vector2int Size2x1 = new Vector2int(2, 1);
+        private static readonly Vector2int Size2x2 = new Vector2int(2, 2);
+        private static readonly Vector2int Size2x3 = new Vector2int(2, 3);
+        private static readonly Vector2int Size3x1 = new Vector2int(3, 1);
+        private static readonly Vector2int Size3x2 = new Vector2int(3, 2);
```

Comment:
What about custom sizes like 4x2?

Reply:
Have not encountered that yet. But the cache system here should be able to handle it anyways since the check for "4x2" would still be evaluated.  
These pre-fetched sizes are simply to quickly include known sizes in so we can avoid further calls into the "HasRoomFor" method.

The normal fetching and caching should still happen even for something like a 4x2 item.